### ISSUE TITLE
Build Error RHEL 7

### ISSUE THREAD

**Issue by danielvdende** [edenhill/librdkafka#362]
The following build error occurs on RHEL 7, when building the master from source:
`/bin/ld:librdkafka.lds:1: syntax error in VERSION script`
I tried `chmod`-ing the `lds-gen.pl` file to allow it to be executable, but did not help. Full output:

    [packer@lrv00065 librdkafka-master]$ sudo ./configure
    using cache file config.cache
    checking for OS or distribution... ok (Linux)
    checking for C compiler from CC env... failed
    checking for gcc (by command)... ok (cached)
    checking for C++ compiler from CXX env... failed
    checking for C++ compiler (g++)... ok (cached)
    checking executable ld... ok (cached)
    checking executable nm... ok (cached)
    checking executable objdump... ok (cached)
    checking executable strip... ok (cached)
    checking for pkgconfig (by command)... ok (cached)
    checking for install (by command)... ok (cached)
    checking for PIC (by compile)... ok (cached)
    checking for GNU-compatible linker options... ok (cached)
    checking for GNU linker-script ld flag... ok (cached)
    checking for __atomic_32 (by compile)... ok (cached)
    checking for __atomic_64 (by compile)... ok (cached)
    checking for socket (by compile)... ok (cached)
    parsing version '0x00080600'... ok (0.8.6)
    checking for libpthread (by pkg-config)... failed
    checking for libpthread (by compile)... ok
    checking for zlib (by pkg-config)... ok
    checking for librt (by pkg-config)... failed
    checking for librt (by compile)... ok
    checking for nm (by env NM)... ok (cached)
    checking for perl (by command)... ok (cached)
    Generated rdkafka.pc
    Generated Makefile.config
    Generated config.h

    Configuration summary:
    prefix                   /usr/local
    ARCH                     x86_64
    CPU                      generic
    MKL_DISTRO               Linux
    CC                       gcc
    CXX                      g++
    LD                       ld
    NM                       nm
    OBJDUMP                  objdump
    STRIP                    strip
    CPPFLAGS                 -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith
    PKG_CONFIG               pkg-config
    INSTALL                  install
    LIB_LDFLAGS              -shared -Wl,-soname,$(LIBFILENAME)
    LDFLAG_LINKERSCRIPT      -Wl,--version-script=
    RDKAFKA_VERSION_STR      0.8.6
    LIBS                     -lpthread -lz   -lrt
    CFLAGS
    CXXFLAGS                 -Wno-non-virtual-dtor
    SYMDUMPER                $(NM) -D
    exec_prefix              /usr/local
    bindir                   /usr/local/bin
    sbindir                  /usr/local/sbin
    libexecdir               /usr/local/libexec
    datadir                  /usr/local/share
    sysconfdir               /usr/local/etc
    sharedstatedir           /usr/local/com
    localstatedir            /usr/local/var
    libdir                   /usr/local/lib
    includedir               /usr/local/include
    infodir                  /usr/local/info
    mandir                   /usr/local/man 
    Generated config.cache
    Now type 'make' to build

    [packer@lrv00065 librdkafka-master]$ sudo make
    make[1]: Entering directory `/home/packer/librdkafka-master/src'
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka.c -o rdkafka.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_broker.c -o rdkafka_broker.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_msg.c -o rdkafka_msg.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_topic.c -o rdkafka_topic.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_defaultconf.c -o rdkafka_defaultconf.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_timer.c -o  rdkafka_timer.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdkafka_offset.c -o rdkafka_offset.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdgz.c -o rdgz.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdaddr.c -o rdaddr.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdrand.c -o rdrand.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdthread.c -o rdthread.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdqueue.c -o rdqueue.o
    gcc -MD -MP -g -O2 -fPIC -Wall -Wfloat-equal -Wpointer-arith  -c rdlog.c -o rdlog.o
    Creating shared library librdkafka.so.1
    gcc  -shared -Wl,-soname,librdkafka.so.1 -Wl,--version-script=librdkafka.lds rdkafka.o rdkafka_broker.o rdkafka_msg.o rdkafka_topic.o rdkafka_defaultconf.o rdkafka_timer.o   rdkafka_offset.o rdcrc32.o rdgz.o rdaddr.o rdrand.o rdthread.o rdqueue.o rdlog.o snappy.o -o     librdkafka.so.1 -lpthread -lz   -lrt
    /bin/ld:librdkafka.lds:1: syntax error in VERSION script
    collect2: error: ld returned 1 exit status
    make[1]: *** [librdkafka.so.1] Error 1
    make[1]: Leaving directory `/home/packer/librdkafka-master/src'
    make: *** [libs] Error 2

**Comment by edenhill** [comment]
Can't reproduce this on master and RHEL7 with gcc-4.8:
https://build.opensuse.org/package/live_build_log/home:edenhill/librdkafka/RHEL_7/x86_64

What gcc version are you using?
Can you paste the output of src/librdkafka.lds?

**Comment by danielvdende** [comment]
@edenhill 
gcc version: `gcc version 4.8.3 20140911 (Red Hat 4.8.3-9) (GCC)`

The librdkafka.lds file is missing, I guess that the error is caused by an issue in the lds-gen.pl file?

**Comment by edenhill** [comment]
Can you try generating the linker script by hand?
`cat src/rdkafka.h | ./lds-gen.pl` 
Does that seem to work?

Also check that WITH_LDS is set in Makefile.config:
`grep WITH_LDS Makefile.config`

**Comment by danielvdende** [comment]
As a workaround (as you suggested @edenhill), I commented WITH_LDS for now. This is the version I will use for my production environment. As a test, I then cloned the repo from github again, following the exact same steps as before. For some reason, the build now succeeds. I checked the `Makefile.config`, and `WITH_LDS` is enabled...

So, my only 2 guesses: Something changed in the repo over the weekend (which seems unlikely if I look at the commit log), or somehow the clone I did previously was corrupted.
